crates/typos-cli/src/file.rs

```diff
@@ -650,10 +642,24 @@ fn is_fixable(typo: &typos::Typo<'_>) -> bool {
     extract_fix(typo).is_some()
 }
 
-fn fix_buffer(mut buffer: Vec<u8>, typos: impl Iterator<Item = typos::Typo<'static>>) -> Vec<u8> {
+fn fix_buffer(
+    mut buffer: Vec<u8>,
+    typos: Vec<typos::Typo<'_>>,
+    correction_indices: Option<Vec<usize>>,
```

Comment:
I feel like instead of parallel vec's, we should just have one iterator with the typo and index in a tuple.  For the existing cases, we could pass in `0`.